gemac/txEngine.py

```diff
@@ -1,24 +1,180 @@
+    0xf9b9df6f, 0x8ebeeff9, 0x17b7be43, 0x60b08ed5, 0xd6d6a3e8, 0xa1d1937e,
+    0x38d8c2c4, 0x4fdff252, 0xd1bb67f1, 0xa6bc5767, 0x3fb506dd, 0x48b2364b,
+    0xd80d2bda, 0xaf0a1b4c, 0x36034af6, 0x41047a60, 0xdf60efc3, 0xa867df55,
+    0x316e8eef, 0x4669be79, 0xcb61b38c, 0xbc66831a, 0x256fd2a0, 0x5268e236,
+    0xcc0c7795, 0xbb0b4703, 0x220216b9, 0x5505262f, 0xc5ba3bbe, 0xb2bd0b28,
+    0x2bb45a92, 0x5cb36a04, 0xc2d7ffa7, 0xb5d0cf31, 0x2cd99e8b, 0x5bdeae1d,
+    0x9b64c2b0, 0xec63f226, 0x756aa39c, 0x026d930a, 0x9c0906a9, 0xeb0e363f,
+    0x72076785, 0x05005713, 0x95bf4a82, 0xe2b87a14, 0x7bb12bae, 0x0cb61b38,
+    0x92d28e9b, 0xe5d5be0d, 0x7cdcefb7, 0x0bdbdf21, 0x86d3d2d4, 0xf1d4e242,
+    0x68ddb3f8, 0x1fda836e, 0x81be16cd, 0xf6b9265b, 0x6fb077e1, 0x18b74777,
+    0x88085ae6, 0xff0f6a70, 0x66063bca, 0x11010b5c, 0x8f659eff, 0xf862ae69,
+    0x616bffd3, 0x166ccf45, 0xa00ae278, 0xd70dd2ee, 0x4e048354, 0x3903b3c2,
+    0xa7672661, 0xd06016f7, 0x4969474d, 0x3e6e77db, 0xaed16a4a, 0xd9d65adc,
+    0x40df0b66, 0x37d83bf0, 0xa9bcae53, 0xdebb9ec5, 0x47b2cf7f, 0x30b5ffe9,
+    0xbdbdf21c, 0xcabac28a, 0x53b39330, 0x24b4a3a6, 0xbad03605, 0xcdd70693,
+    0x54de5729, 0x23d967bf, 0xb3667a2e, 0xc4614ab8, 0x5d681b02, 0x2a6f2b94,
+    0xb40bbe37, 0xc30c8ea1, 0x5a05df1b, 0x2d02ef8d)
+""" Feedback terms Table for CRC32
+
+The feedback terms table consists of 256, 32-bit entries used for Table
+Driven Implementation of CRC32.
+
+References:
+    http://opensource.apple.com//source/xnu/xnu-1456.1.26/bsd/libkern/crc32.c
+    http://www.zlib.net/crc_v3.txt
+
+Specification of crc algorithm used to calculate Frame Check Sequence
+   Name   : "CRC-32"
+   Width  : 32
+   Poly   : 04C11DB7
+   Init   : FFFFFFFF
+   RefIn  : True
+   RefOut : True
+   XorOut : FFFFFFFF
+   Check  : CBF43926
+
+Note:
+    CRC output from the implementation of the algorithm can be compared to
+    output of CRC32-Reversed-LitleEndian by providing hex input at:
+    http://www.scadacore.com/field-applications/programming-calculators/online-checksum-calculator/
+"""
 
 
 @block
-def txEngine(gtxclk, txd2engine, txd2gmii, txpause, sampled_pauseval):
+def txengine(txclientintf, txgmii_intf, txflowintf, txconfig, reset):
+    """Transmit Engine.
 
-    """
-    *Transmit Engine*
     Accepts Ethernet frame data from the Client Transmitter interface,
     adds preamble to the start of the frame, add padding bytes and frame
     check sequence. It ensures that the inter-frame spacing between successive
     frames is at least the minimum specified. The frame is then converted
     into a format that is compatible with the GMII and sent to the GMII Block.
 
-    gtxClk - Reference Clock for Transmit Operations from Host
-    txData2Engine - Transmit Data Channel From Client
-    txData2GMII - Transmit Data Channel To GMII
-    transmitPauseFrame - Control Signal from FLowControl to Indicate
-                            Sending Pause Control Frame
-    sampledPausesVal
+    Args:
```

Comment:
I should take care of it. I normally am using <name>intf , ex. txclientintf. I only used <name>_intf
in txgmii_intf bcoz of presence of two i's in gmii causing inconvenience.